File: Java/src/main/java/com/gildedrose/Normal.java
package com.gildedrose;

public class Normal extends Quality {

    @Override
    public void update(Item item) {
        decreaseQuality(item);

        updateSellIn(item);
    }

    @Override
    protected void expires(Item item) {
        decreaseQuality(item);
    }

    @Override
    public boolean applies(Item item) {
        return true;
    }
}

File: go/itembehaviour/normal.go
package itembehaviour

import "github.com/davidmachadomeli/Gilded-Rose-Kata/go/item"

// item normal
type Normal struct {}

func (n *Normal) Update(item *item.Item) {
	item.SellIn -= 1

	item.Quality -= 1

	if item.SellIn < 0 {
		item.Quality -= 1
	}

	item.Quality = LimitTo0(item.Quality)
}
